packages/angular_devkit/build_angular/src/babel/plugins/adjust-angular-classes.ts

```diff
@@ -0,0 +1,300 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { NodePath, PluginObj, PluginPass, types } from '@babel/core';
+import annotateAsPure from '@babel/helper-annotate-as-pure';
+
+/**
+ * The name of the Typescript decorator helper function created by the TypeScript compiler.
+ */
+const TSLIB_DECORATE_HELPER_NAME = '__decorate';
+
+/**
+ * The set of Angular static fields that should always be wrapped.
+ * These fields may appear to have side effects but are safe to remove if the associated class
+ * is otherwise unused within the output.
+ */
+const angularStaticsToWrap = new Set([
+  'ɵcmp',
+  'ɵdir',
+  'ɵfac',
+  'ɵinj',
+  'ɵmod',
+  'ɵpipe',
+  'ɵprov',
+  'INJECTOR_KEY',
+]);
+
+/**
+ * An object map of static fields and related value checks for discovery of Angular generated
+ * JIT related static fields.
+ */
+const angularStaticsToElide: Record<string, (path: NodePath<types.Expression>) => boolean> = {
+  'ctorParameters'(path) {
+    return path.isFunctionExpression() || path.isArrowFunctionExpression();
+  },
+  'decorators'(path) {
+    return path.isArrayExpression();
+  },
+  'propDecorators'(path) {
+    return path.isObjectExpression();
+  },
+};
+
+/**
+ * Provides one or more keywords that if found within the content of a source file indicate
+ * that this plugin should be used with a source file.
+ *
+ * @returns An a string iterable containing one or more keywords.
+ */
+export function getKeywords(): Iterable<string> {
+  return ['class'];
+}
+
+/**
+ * Analyze the sibling nodes of a class to determine if any downlevel elements should be
+ * wrapped in a pure annotated IIFE. Also determines if any elements have potential side
+ * effects.
+ *
+ * @param origin The starting NodePath location for analyzing siblings.
+ * @param classIdentifier The identifier node that represents the name of the class.
+ * @param allowDecorators Whether to allow decorators to be wrapped.
+ * @returns An object containing the results of the analysis.
+ */
+function analyzeClassSiblings(
+  origin: NodePath,
+  classIdentifier: types.Identifier,
+  allowDecorators: boolean,
```

Comment:
Maybe, a more self-explanatory name would be `wrapDecorators` or `allowWrappingDecorators`